commit 035998112737e52cb229e342913ef404e5a51040
Author: Eric Blake <ebb9@byu.net>
Date:   Thu Nov 22 07:34:32 2007 -0700

    Security fix: avoid arbitrary code execution with 'm4 -F'.
    
    * src/freeze.c (produce_frozen_state): Never pass raw file name as
    printf format.
    * NEWS: Document this fix.
    
    Signed-off-by: Eric Blake <ebb9@byu.net>
    (cherry picked from commit 031a71a80442ed2ad3c2ee14d5811c786a12c51b)

diff --git a/src/freeze.c b/src/freeze.c
index 3363a14..ccdccf4 100644
--- a/src/freeze.c
+++ b/src/freeze.c
@@ -58,7 +58,7 @@ produce_frozen_state (const char *name)
 
   if (file = fopen (name, O_BINARY ? "wb" : "w"), !file)
     {
-      M4ERROR ((warning_status, errno, name));
+      M4ERROR ((warning_status, errno, "%s", name));
       return;
     }
 
